pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +814,136 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+// UpdateTTL Call clickhouse-client to send ttl change request
+func (fa *flowAggregator) UpdateTTL(ttl string) {
+	if fa.dbExportProcess == nil {
+		klog.ErrorS(nil, "ClickHouse hasn't been enabled yet")
+		return
+	}
+	if err := fa.dbExportProcess.UpdateTTL(ttl); err != nil {
+		klog.ErrorS(nil, "Error when updating ttl")
+		return
+	}
+}
+
+// updateIfConfigChanges has a watcher to detect if modification happens. Then check whether we should update clickhouse or flow-collector at runtime
+func (fa *flowAggregator) updateIfConfigChanges(stopCh <-chan struct{}) {
+	if fa.configMap == nil {
+		var err error
+		fa.configMap, err = flowaggregatorconfig.GetFAConfigMap(fa.k8sClient)
+		if err != nil {
+			klog.ErrorS(err, "Get flow aggregator configmap failed:")
+		}
+	}
+	go wait.NonSlidingUntil(fa.watchConfigmap, 5*time.Second, stopCh)
+}
+
+func (fa *flowAggregator) watchConfigmap() {
+
+	watcher, err := fa.k8sClient.CoreV1().ConfigMaps("flow-aggregator").Watch(context.TODO(), metav1.SingleObject(metav1.ObjectMeta{
+		Name: os.Getenv("FA_CONFIG_MAP_NAME"), Namespace: "flow-aggregator"}))
+	if err != nil {
+		klog.ErrorS(err, "Error when creating watcher")
+	}
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if event.Type != watch.Modified {
+				continue
+			}
+			if !ok {
+				return
+			}
+			klog.Info("Event for configuration file")
+			configMap, err := flowaggregatorconfig.GetFAConfigMap(fa.k8sClient)
```

Comment:
I am not sure why we get the ConfigMap from the API here? The new ConfigMap should already be stored in `event.Obejct`.

Reply:
Fixed. I forgot to changed it.